x/gamm/pool-models/stableswap/stableswap_pool.go

```diff
@@ -0,0 +1,29 @@
+package stableswap
+
+import (
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v7/x/gamm/types"
+)
+
+var _ types.PoolI = &Pool{}
+
+// NewStableswapPool returns a stableswap pool
+// Invariants that are assumed to be satisfied and not checked:
+// * len(initialLiquidity) = 2
+// * FutureGovernor is valid
+// * poolID doesn't already exist
+func NewStableswapPool(poolId uint64, stableswapPoolParams PoolParams, initialLiquidity sdk.Coins, futureGovernor string, blockTime time.Time) (Pool, error) {
+	pool := &Pool{
```

Comment:
lets remove the reference taken here, and dereference taken on on line 28?

Reply:
Should be added in next commit!